src/systemctl/systemctl-list-unit-files.c

```diff
@@ -3,33 +3,43 @@
 #include "ansi-color.h"
 #include "bus-error.h"
 #include "bus-locator.h"
+#include "path-util.h"
 #include "sort-util.h"
 #include "systemctl-list-unit-files.h"
 #include "systemctl-util.h"
 #include "systemctl.h"
 #include "terminal-util.h"
 
 static int compare_unit_file_list(const UnitFileList *a, const UnitFileList *b) {
-        const char *d1, *d2;
+        const char *d1, *d2, *a_id = NULL, *b_id = NULL;
+        int u, v;
 
         d1 = strrchr(a->path, '.');
         d2 = strrchr(b->path, '.');
 
         if (d1 && d2) {
-                int r;
-
-                r = strcasecmp(d1, d2);
+                int r = strcasecmp(d1, d2);
```

Comment:
`int r`, is special to us, because it is the primary place we store func return codes in. Hence please define a outermost func scope, always.

only `r` gets that special treatmen